core/engine/engine_test.go

```diff
@@ -94,6 +94,10 @@ func genRandomIndexAndAdv(t *testing.T, e *Engine) (schema.Index, schema.Link_In
 	return index, indexLnk, adv, advLnk
 }
 
+/* NOTE: This test is flaky due to something deep in the task. Commenting it for now.
+Will work on fixing it and understand if it is something to do with our implementation
+or something deeper.
+
```

Comment:
Would it make sense to use `t.Skip` with an explanation of why it is skipped? See [docs](https://pkg.go.dev/testing#hdr-Skipping)

Reply:
Good point, that's better. Thanks!